include/modem/modem_info.h

```diff
@@ -57,14 +48,70 @@ extern "C" {
 /** SNR offset value. */
 #define SNR_OFFSET_VAL 24
 
-/** Modem returns RSRP and RSRQ as index values which require
- * a conversion to dBm and dB respectively. See modem AT
- * command reference guide for more information.
+/** @brief Converts RSRP index value returned by the modem to dBm.
+ *
+ * The index value of RSRP can be converted to decibel with the following formula:
+ * * Index < 0: Index – 140
+ * * Index = 0: Not used
+ * * Index > 0: Index – 141
+ *
+ * Example values:
+ * * -17: RSRP < -156 dBm
+ * * -16: -156 ≤ RSRP < -155 dBm
+ * * ...
+ * * -3: -143 ≤ RSRP < -142 dBm
+ * * -2: -142 ≤ RSRP < -141 dBm
+ * * -1: -141 ≤ RSRP < -140 dBm
+ * * 0: Not used.
+ * * 1: -140 ≤ RSRP < -139 dBm
+ * * 2: -139 ≤ RSRP < -138 dBm
+ * * ...
+ * * 95: -46 ≤ RSRP < -45 dBm
+ * * 96: -45 ≤ RSRP < -44 dBm
+ * * 97: -44 ≤ RSRP dBm
+ *
+ * There are use cases where the index value 0 is used to represent RSRP < -140 dBm.
+ *
+ * See modem AT command reference guide for more information.
  */
-#define RSRP_IDX_TO_DBM(rsrp) ((rsrp) - RSRP_OFFSET_VAL)
-
-#define RSRQ_IDX_TO_DB(rsrq) ((((float)(rsrq)) * RSRQ_SCALE_VAL) - \
-			      RSRQ_OFFSET_VAL)
+#define RSRP_IDX_TO_DBM(rsrp) ((rsrp) < 0 ?  (rsrp) - 140 : (rsrp) - 141)
+
+/** @brief Converts RSRQ index value returned by the modem to dB.
+ *
+ * The index value of RSRQ can be converted to decibel with the following formula:
```

Comment:
Should "decibel" be plural or not, @peknis?

Reply:
Waiting for Jaakko's feedback

Reply:
Can be either.

Reply:
Changing this to dB to be aligned with previous macro